NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -309,11 +320,26 @@ public boolean hasAccess(String groupname, UUID player, PermissionType perm) {
 			NameLayerPlugin.getInstance().getLogger().log(Level.INFO, "hasAccess failed (access denied), could not find group " + groupname);
 			return false;
 		}
-		
 		GroupPermission perms = getPermissionforGroup(group);
-		PlayerType rank = group.getPlayerType(player);	
-		
-		return perms.isAccessible(rank, perm);
+		for(PlayerType rank : getRecursivePlayerTypes(group, player)) {
+			if (perms.hasPermission(rank, perm)) {
+				//player has right rank in the group itself or at least one super group
+				return true;
+			}
+		}		
+		return false;
+	}
+	
+	private List<PlayerType> getRecursivePlayerTypes(Group group, UUID player) {
```

Comment:
Eh, for what its worth I starting writing out a counter, and realized a flaw; since players can "leave" a group at any time, they could easily force the supergroup acl to apply to them simply by leaving the subgroup if previously in both.

Oh well.